### PR TITLE
chore(CI): delete needless link when build packages

### PR DESCRIPTION
**Pull request by Rory-Z**

### CODE REVIEW

File: deploy/docker/Dockerfile [hunk 1]
```diff
@@ -26,7 +26,9 @@ COPY . /emqx
 ARG PKG_VSN
 ARG EMQX_NAME=emqx
 
-RUN cd /emqx && make $EMQX_NAME
+RUN cd /emqx \
+    && rm -rf _build/$EMQX_NAME/lib \
```
Comment (zmstone) [inline deploy/docker/Dockerfile:30]: delete the entire profile ?

File: .github/workflows/build_packages.yaml [hunk 2]
```diff
@@ -169,9 +169,11 @@ jobs:
     - name: build
       run: |
         . $HOME/.kerl/${{ matrix.erl_otp }}/activate
-        make -C source ensure-rebar3
-        sudo cp source/rebar3 /usr/local/bin/rebar3
-        make -C source ${{ matrix.profile }}-zip
+        cd source
+        make ensure-rebar3
+        sudo cp rebar3 /usr/local/bin/rebar3
+        rm -rf _build/${{ matrix.profile }}/lib
```
Comment (zmstone) [inline .github/workflows/build_packages.yaml:175]: delete the profile all together ?

File: deploy/docker/Dockerfile (hunk 1 above)
Comment (Rory-Z) [reply deploy/docker/Dockerfile:30]: Other profiles will not be compiled or copied to the distribution, so I don't think it's necessary to delete

File: .github/workflows/build_packages.yaml (hunk 2 above)
Comment (Rory-Z) [reply .github/workflows/build_packages.yaml:175]: Other profiles will not be compiled or copied to the distribution, so I don't think it's necessary to delete

**Pull request merged**